### ISSUE TITLE
Ctrl+C not captured when waiting on Console.ReadKey

### ISSUE THREAD

**Issue by jamesfoster** [Cysharp/ConsoleAppFramework#82]
I'm building an interactive console app where arguments can be specified on the command line, or the user will be asked to enter the argument values at run-time.

According to the docs, ConsoleAppFramework handles the cancel event? I've set the shutdown time to zero so I'd expect the app to terminate immediately on Ctrl+C even tho we're waiting on the user.

I've tried handling Ctrl+C myself by setting `Console.TreatControlCAsInput = true` and checking the returned key info. This seems to work in debug but not when running in release mode.

Any help would be appreciated.

**Comment by jamesfoster** [comment]
The sample you sent works. Not sure what it is about what I'm doing that's causing it. This may be running in a different thread to the main thread but I don't think it's that. I tried the following to test but this also works ok. It's unfortunate that there's no version of ReadKey that takes a `CancellationToken`

```
ConsoleApp.Run(args, (ConsoleAppContext ctx) =>
{
    return Task.Run(async () =>
    {
        while (!ctx.CancellationToken.IsCancellationRequested)
        {
            ...
```

**Comment by jamesfoster** [comment]
Is there any way to disable the frameworks handling of Ctrl+C? It seems that when using `TreatControlCAsInput` it breaks Ctrl+V to paste.

Without using the framework, the following works as expected:

```
public static void Main()
{
  while (true)
  {
    var key = Console.ReadKey(intercept: true);

    Console.Write(key.KeyChar);
  }
}
```

**Comment by neuecc** [comment]
I've released v4.2.1, it can write following

```csharp
// case of Console.Read/ReadLine, pressed Ctrl+C, Read returns null.
ConsoleApp.Run(args, (ConsoleAppContext ctx) =>
{
    var read = Console.ReadLine();
    if (read == null) ctx.Terminate();
});

// case of Console.ReadKey, can not cancel itself so use with Task.Run and WaitAsync.
ConsoleApp.Run(args, async (ConsoleAppContext ctx) =>
{
    var key = await Task.Run(() => Console.ReadKey()).WaitAsync(ctx.CancellationToken);
});
```

**Comment by jamesfoster** [comment]
Thanks, I'll check it out.
